pkg/solana/logpoller/filters.go

```diff
@@ -119,9 +119,26 @@ func (fl *filters) RegisterFilter(ctx context.Context, filter Filter) error {
 		if existingFilter.IsBackfilled {
 			// if existing filter was already backfilled, but starting block was higher we need to backfill filter again
 			filter.IsBackfilled = existingFilter.StartingBlock <= filter.StartingBlock
+
+			// also trigger a backfill if IncludeReverted is being updated from false to true
+			if !existingFilter.IncludeReverted && filter.IncludeReverted {
+				filter.IsBackfilled = false
+			}
 		}
 
 		fl.removeFilterFromIndexes(*existingFilter)
+	} else {
+		// ensure that the value of IncludeReverted isn't different from any other filters with the same address and event type
```

Comment:
We'll end up with two similar filters and different IncludeReverted, if there were two filters and I've changed IncludeReverted for one of them.